src/Config/Validator/Deploy/MageModeVariable.php

```diff
@@ -0,0 +1,68 @@
+<?php
+/**
+ * Copyright © Magento, Inc. All rights reserved.
+ * See COPYING.txt for license details.
+ */
+declare(strict_types=1);
+
+namespace Magento\MagentoCloud\Config\Validator\Deploy;
+
+use Magento\MagentoCloud\App\Error;
+use Magento\MagentoCloud\Config\EnvironmentDataInterface;
+use Magento\MagentoCloud\Config\Validator;
+use Magento\MagentoCloud\Config\Validator\ResultFactory;
+use Magento\MagentoCloud\Config\ValidatorInterface;
+
+/**
+ * Validates value of MAGE_MODE variable.
+ */
+class MageModeVariable implements ValidatorInterface
+{
+    public const PRODUCTION_MODE = 'production';
+
+    /**
+     * @var EnvironmentDataInterface
+     */
+    private $envData;
+
+    /**
+     * @var ResultFactory
+     */
+    private $resultFactory;
+
+    /**
+     * @param EnvironmentDataInterface $envData
+     * @param ResultFactory $resultFactory
+     */
+    public function __construct(
+        EnvironmentDataInterface $envData,
+        ResultFactory $resultFactory
+    ) {
+        $this->envData = $envData;
+        $this->resultFactory = $resultFactory;
+    }
+
+    /**
+     * @return Validator\ResultInterface
+     */
+    public function validate(): Validator\ResultInterface
+    {
+        $mageMode = $this->envData->getMageMode();
+        if (!$mageMode || $mageMode == self::PRODUCTION_MODE) {
+            return $this->resultFactory->success();
+        }
+
+        return $this->resultFactory->error(
+            sprintf(
+                'Environment variable MAGE_MODE was found and the value differs from "%s".',
```

Comment:
Don't you want to put these message and suggestion into `cloud.error.yaml` file and call here?
```
return $this->resultFactory->errorByCode(Error::WARN_NOT_SUPPORTED_MAGE_MODE);
```
As this message/suggestion is static we can use them from `cloud.error.yaml` to avoid message duplication.

Reply:
you mean schema.error.yaml? I've added suggestion there, it is almost the same

Reply:
Yes schema.error.yaml, sorry. I mean you can remove message and suggestion from the code and use them from schema.error.yaml

Reply:
hm. That's a good point. But I just thought that that error may be a general and we may use in another error message with wrong mage mode. Or there is not sense in it. What do you think?

Reply:
Looks like we have only one possible variant, so I think there are no sense to n it.